osu.Game/Skinning/SkinInfo.cs

```diff
@@ -52,13 +55,15 @@ public virtual Skin CreateInstance(IStorageResourceProvider resources)
                 // for user modified skins. This aims to amicably handle that.
                 // If we ever add more default skins in the future this will need some kind of proper migration rather than
                 // a single fallback.
-                return new TrianglesSkin(this, resources);
+                return typeof(TrianglesSkin);
             }
 
-            return (Skin)Activator.CreateInstance(type, this, resources)!;
+            return type;
         }
 
-        public IList<RealmNamedFileUsage> Files { get; } = null!;
+        public virtual Skin CreateInstance(IStorageResourceProvider resources) => (Skin)Activator.CreateInstance(GetInstanceType(), this, resources)!;
+
+        public virtual IList<RealmNamedFileUsage> Files { get; } = null!;
```

Comment:
I'm surprised realm even allows you to make this `virtual`.... Seems like a very wrong direction.

Let's not, unless there's another case we've done this.

Reply:
Seems like I don't even need it. The usage was inserting dummy realm files in `TestSceneSkinMigration.TestSkinInfo`, and removing the override does not seem to have any effect on the test results.